### ISSUE TITLE
Unhandled exception has occured/Assestion Error/ValueError: too many values to unpack

### ISSUE THREAD

**Issue by threeFatCat** [kiliman/operator-mono-lig#54]
Hi! I'm having trouble generating fonts with ligature. I tried to use both Python 2 and Python 3 and I get errors in both. I also tried using the build.bat and build.sh but still having the same output. Here is the output:

`C:\Users\ricadiz\operator-mono-lig-master>build
Building OperatorMonoLig-Medium
Dumping ".\original\OperatorMono-Medium.otf" to ".\original\OperatorMono-Medium.ttx"...
Dumping 'GlyphOrder' table...
Dumping 'head' table...
Dumping 'hhea' table...
Dumping 'maxp' table...
Dumping 'OS/2' table...
Dumping 'name' table...
Dumping 'cmap' table...
Dumping 'post' table...
Dumping 'CFF ' table...
Dumping 'GPOS' table...
Dumping 'GSUB' table...
Dumping 'hmtx' table...
Dumping 'DSIG' table...
Reading original font file ./original/OperatorMono-Medium.ttx
Patching names
Patching glyphs
Patching gpos
Patching gsub
Patching hmtx
Patching charstrings
Writing ligature font file ./build/OperatorMonoLig-Medium.ttx
Done
Compiling ".\build\OperatorMonoLig-Medium.ttx" to ".\build\OperatorMonoLig-Medium.otf"...
Parsing 'GlyphOrder' table...
Parsing 'head' table...
Parsing 'hhea' table...
Parsing 'maxp' table...
Parsing 'OS/2' table...
Parsing 'name' table...
Parsing 'cmap' table...
Parsing 'post' table...
Parsing 'CFF ' table...
Parsing 'GPOS' table...
Parsing 'GSUB' table...
Parsing 'hmtx' table...
Parsing 'DSIG' table...
ERROR: Unhandled exception has occurred
Traceback (most recent call last):
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 399, in main
    process(jobs, options)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 373, in process
    action(input, output, options)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\loggingTools.py", line 372, in wrapper
    return func(*args, **kwds)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 291, in ttCompile
    ttf.save(output)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 166, in save
    writer_reordersTables = self._save(tmp)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 205, in _save
    self._writeTable(tag, writer, done, tableCache)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 618, in _writeTable
    self._writeTable(masterTable, writer, done, tableCache)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 622, in _writeTable
    tabledata = self.getTableData(tag)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 640, in getTableData
    return self.tables[tag].compile(self)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\tables\C_F_F_.py", line 20, in compile
    self.cff.compile(f, otFont, isCFF2=False)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\cffLib\__init__.py", line 108, in compile
    topDict.recalcFontBBox()
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\cffLib\__init__.py", line 2340, in recalcFontBBox
    bounds = charString.calcBounds(self.CharStrings)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 984, in calcBounds
    self.draw(boundsPen)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 979, in draw
    extractor.execute(self)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 301, in execute
    rv = handler(index)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 340, in op_callsubr
    self.execute(subr)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 301, in execute
    rv = handler(index)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 360, in op_hintmask
    hintMaskBytes, index = self.callingStack[-1].getBytes(index, self.hintMaskBytes)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 1073, in getBytes
    assert len(bytes) == nBytes
AssertionError
(Hit any key to exit)
Building OperatorMonoLig-MediumItalic
Dumping ".\original\OperatorMono-MediumItalic.otf" to ".\original\OperatorMono-MediumItalic.ttx"...
Dumping 'GlyphOrder' table...
Dumping 'head' table...
Dumping 'hhea' table...
Dumping 'maxp' table...
Dumping 'OS/2' table...
Dumping 'name' table...
Dumping 'cmap' table...
Dumping 'post' table...
Dumping 'CFF ' table...
Dumping 'GPOS' table...
Dumping 'GSUB' table...
Dumping 'hmtx' table...
Dumping 'DSIG' table...
Reading original font file ./original/OperatorMono-MediumItalic.ttx
Patching names
Patching glyphs
Patching gpos
Patching gsub
Patching hmtx
Patching charstrings
Writing ligature font file ./build/OperatorMonoLig-MediumItalic.ttx
Done
Compiling ".\build\OperatorMonoLig-MediumItalic.ttx" to ".\build\OperatorMonoLig-MediumItalic.otf"...
Parsing 'GlyphOrder' table...
Parsing 'head' table...
Parsing 'hhea' table...
Parsing 'maxp' table...
Parsing 'OS/2' table...
Parsing 'name' table...
Parsing 'cmap' table...
Parsing 'post' table...
Parsing 'CFF ' table...
Parsing 'GPOS' table...
Parsing 'GSUB' table...
Parsing 'hmtx' table...
Parsing 'DSIG' table...
ERROR: Unhandled exception has occurred
Traceback (most recent call last):
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 399, in main
    process(jobs, options)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 373, in process
    action(input, output, options)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\loggingTools.py", line 372, in wrapper
    return func(*args, **kwds)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 291, in ttCompile
    ttf.save(output)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 166, in save
    writer_reordersTables = self._save(tmp)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 205, in _save
    self._writeTable(tag, writer, done, tableCache)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 618, in _writeTable
    self._writeTable(masterTable, writer, done, tableCache)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 622, in _writeTable
    tabledata = self.getTableData(tag)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 640, in getTableData
    return self.tables[tag].compile(self)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\tables\C_F_F_.py", line 20, in compile
    self.cff.compile(f, otFont, isCFF2=False)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\cffLib\__init__.py", line 108, in compile
    topDict.recalcFontBBox()
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\cffLib\__init__.py", line 2340, in recalcFontBBox
    bounds = charString.calcBounds(self.CharStrings)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 984, in calcBounds
    self.draw(boundsPen)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 979, in draw
    extractor.execute(self)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 301, in execute
    rv = handler(index)
  File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 578, in op_endchar
    adx, ady, bchar, achar = args
ValueError: too many values to unpack
(Hit any key to exit)`

Do I need to place the whole Operator Mono font family in the original directory? or just the font that I will use?

By the way, I'm using the following:

ttx --version
3.25.0
node --version
v9.1.0

Windows 10 x64

**Comment by Orillian** [comment]
`C:\Users\ricadiz\operator-mono-lig-master>build Building OperatorMonoLig-Medium 
Dumping ".\original\OperatorMono-Medium.otf" to ".\original\OperatorMono-Medium.ttx"... 
Dumping 'GlyphOrder' table... 
Dumping 'head' table... 
Dumping 'hhea' table... 
Dumping 'maxp' table... 
Dumping 'OS/2' table... 
Dumping 'name' table... 
Dumping 'cmap' table... 
Dumping 'post' table... 
Dumping 'CFF ' table... 
Dumping 'GPOS' table... 
Dumping 'GSUB' table... 
Dumping 'hmtx' table... 
Dumping 'DSIG' table... 
Reading original font file ./original/OperatorMono-Medium.ttx 
Patching names 
Patching glyphs 
Patching gpos 
Patching gsub 
Patching hmtx 
Patching charstrings 
Writing ligature font file ./build/OperatorMonoLig-Medium.ttx 
Done 
Compiling ".\build\OperatorMonoLig-Medium.ttx" to ".\build\OperatorMonoLig-Medium.otf"... 
Parsing 'GlyphOrder' table... 
Parsing 'head' table... 
Parsing 'hhea' table... 
Parsing 'maxp' table... 
Parsing 'OS/2' table... 
Parsing 'name' table... 
Parsing 'cmap' table... 
Parsing 'post' table... 
Parsing 'CFF ' table... 
Parsing 'GPOS' table... 
Parsing 'GSUB' table... 
Parsing 'hmtx' table... 
Parsing 'DSIG' table... ERROR: Unhandled exception has occurred Traceback (most recent call last): 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 399, in main process(jobs, options) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 373, in process action(input, output, options) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\loggingTools.py", line 372, in wrapper return func(*args, **kwds) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 291, in ttCompile ttf.save(output) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 166, in save writer_reordersTables = self._save(tmp) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 205, in _save self._writeTable(tag, writer, done, tableCache) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 618, in _writeTable self._writeTable(masterTable, writer, done, tableCache) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 622, in _writeTable tabledata = self.getTableData(tag) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 640, in getTableData return self.tables[tag].compile(self) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\tables\C_F_F_.py", line 20, in compile self.cff.compile(f, otFont, isCFF2=False) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\cffLib\__init__.py", line 108, in compile topDict.recalcFontBBox()
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\cffLib\__init__.py", line 2340, in recalcFontBBox bounds = charString.calcBounds(self.CharStrings) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 984, in calcBounds self.draw(boundsPen) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 979, in draw extractor.execute(self) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 301, in execute rv = handler(index) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 340, in op_callsubr self.execute(subr) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 301, in execute rv = handler(index) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 360, in op_hintmask hintMaskBytes, index = self.callingStack[-1].getBytes(index, self.hintMaskBytes) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 1073, in getBytes assert len(bytes) == nBytes AssertionError (Hit any key to exit) 

Building OperatorMonoLig-MediumItalic 
Dumping ".\original\OperatorMono-MediumItalic.otf" to ".\original\OperatorMono-MediumItalic.ttx"... 
Dumping 'GlyphOrder' table... 
Dumping 'head' table... 
Dumping 'hhea' table... 
Dumping 'maxp' table... 
Dumping 'OS/2' table... 
Dumping 'name' table... 
Dumping 'cmap' table... 
Dumping 'post' table... 
Dumping 'CFF ' table... 
Dumping 'GPOS' table... 
Dumping 'GSUB' table... 
Dumping 'hmtx' table... 
Dumping 'DSIG' table... 
Reading original font file ./original/OperatorMono-MediumItalic.ttx 
Patching names 
Patching glyphs 
Patching gpos 
Patching gsub 
Patching hmtx 
Patching charstrings 
Writing ligature font file ./build/OperatorMonoLig-MediumItalic.ttx 
Done 
Compiling ".\build\OperatorMonoLig-MediumItalic.ttx" to ".\build\OperatorMonoLig-MediumItalic.otf"... 
Parsing 'GlyphOrder' table... 
Parsing 'head' table... 
Parsing 'hhea' table... 
Parsing 'maxp' table... 
Parsing 'OS/2' table... 
Parsing 'name' table... 
Parsing 'cmap' table... 
Parsing 'post' table... 
Parsing 'CFF ' table... 
Parsing 'GPOS' table... 
Parsing 'GSUB' table... 
Parsing 'hmtx' table... 
Parsing 'DSIG' table... ERROR: Unhandled exception has occurred Traceback (most recent call last): 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 399, in main process(jobs, options) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 373, in process action(input, output, options) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\loggingTools.py", line 372, in wrapper return func(*args, **kwds) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttx.py", line 291, in ttCompile ttf.save(output) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 166, in save writer_reordersTables = self._save(tmp) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 205, in _save self._writeTable(tag, writer, done, tableCache) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 618, in _writeTable self._writeTable(masterTable, writer, done, tableCache) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 622, in _writeTable tabledata = self.getTableData(tag) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\ttFont.py", line 640, in getTableData return self.tables[tag].compile(self) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\ttLib\tables\C_F_F_.py", line 20, in compile self.cff.compile(f, otFont, isCFF2=False) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\cffLib\__init__.py", line 108, in compile topDict.recalcFontBBox() 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\cffLib\__init__.py", line 2340, in recalcFontBBox bounds = charString.calcBounds(self.CharStrings) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 984, in calcBounds self.draw(boundsPen) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 979, in draw extractor.execute(self) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 301, in execute rv = handler(index) 
File "d:\python27\lib\site-packages\fonttools-3.25.0-py2.7.egg\fontTools\misc\psCharStrings.py", line 578, in op_endchar adx, ady, bchar, achar = args ValueError: too many values to unpack (Hit any key to exit)`

There...maybe a little easier to read?

Have you tried Python 3.x?

[Some discussion about this error in other fonts.](https://github.com/fonttools/fonttools/issues/1229) It looks like there might be a bit of inconsistency between different versions of python and how fonttools deals with glyph names.

**Comment by kiliman** [comment]
It appears to be an issue with his fonts. `ttx` is the one parsing the OpenType files.

**Comment by threeFatCat** [comment]
Yes, actually I used first Python 3.x but since I'm getting error similar to that, and I was reading the issues here one of @kiliman's [comment ](https://github.com/kiliman/operator-mono-lig/issues/34)is probably an issue with Python 3.x, thus I tried Python 2.

Both version of Python has build output with 0 size .otf and with size .ttx which it looks like unfinished for me.

Perhaps I'm not placing the right fonts? What I did, I just placed one Operator Mono medium and one Operator Mono italic. Both has no spaces in the file name and with dash for the font style, e.g. _OperatorMono-Medium.otf_

Yay, sorry for that format posting I was confused how to format the logs when I posted it here.
